src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -209,24 +212,38 @@ public static ObjectType UnderlyingGraphQLEntityType(IType type)
         }
 
         /// <summary>
-        /// Parse a given string value to supported GraphQL Type and GraphQLValueNode
+        /// Translates a JSON string or number value defined in the runtime configuration to a GraphQL {Type}ValueNode which represents
+        /// the associated GraphQL type. The target value type is referenced from the passed in parameterDefinition which
+        /// holds database schema metadata.
         /// </summary>
-        public static Tuple<string, IValueNode> GetGraphQLTypeAndNodeTypeFromStringValue(string stringValue)
+        /// <param name="defaultValueFromConfig">String representation of default value defined in runtime config.</param>
+        /// <param name="parameterDefinition">Database schema metadata for stored procedure parameter which include value and value type.</param>
+        /// <returns>Tuple where first item is the string representation of a GraphQLType (e.g. "Byte", "Int", "Decimal")
+        /// and the second item is the GraphQL *type*ValueNode </returns>
+        /// <exception cref="DataApiBuilderException">Raised when parameter casting fails due to unsupported type.</exception>
+        public static Tuple<string, IValueNode> ConvertValueToGraphQLType(string defaultValueFromConfig, ParameterDefinition parameterDefinition)
```

Comment:
Should this be TryConvert? The Parse method below might fail if the config has an invalid parameter value - not compatible with the type present for the stored procedure parameter

Reply:
It could either be TryConvert or left as-is. 

A failure here halts engine startup. The issue is either incorrectly written runtime configuration (default parameter values are of incorrect type) or an attempt to create a stored procedure entity when the stored procedure schema contains types that dab engine does not support ( no matter what you set as the parameter value, we'd have to fail).

I think the minimum change here would be to wrap the switch in a try catch to catch any parsing errors and raise a dab exception accordingly.

Reply:
catching parsing errors and providing dab exception sounds good.